src/data/feature_index.js

```diff
@@ -247,6 +249,16 @@ class FeatureIndex {
 
         return false;
     }
+
+    getId(feature: VectorTileFeature, sourceLayerId: string): string | number | void {
+        let id = feature.id;
+        if (this.promoteId) {
+            const propName = typeof this.promoteId === 'string' ? this.promoteId : this.promoteId[sourceLayerId];
```

Comment:
Is the intention to allow a single string that is used as the promoteId for all source-layers?

Reply:
Yes — for better ergonomics if all source layers share the same unique property (which can often be the case), and also to better support GeoJSON sources (no need to set `{_geojsonTileLayer: '...'}`).